mm2src/adex_cli/Cargo.toml

```diff
@@ -23,14 +25,14 @@ mm2_net = { path = "../mm2_net" }
 mm2_number = { path = "../mm2_number" }
 mm2_rpc = { path = "../mm2_rpc"}
 passwords = "3.1"
+rpc = { path = "../mm2_bitcoin/rpc" }
+rustls = { version = "^0.20.4", features = [ "dangerous_configuration" ] }
```

Comment:
cc @DeckerSU @Alrighttt 


@rozhkovdmitrii why two diff versions? (we seem using a total of 3 diff across codebase, `0.19.1`, `0.20.4` and `0.20.8`)

cli lockfile:
```
[[package]]
name = "rustls"
version = "0.19.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "35edb675feee39aec9c99fa5ff985081995a06d594114ae14cbe797ad7b7a6d7"
dependencies = [
 "base64 0.13.1",
 "log 0.4.17",
 "ring",
 "sct 0.6.1",
 "webpki 0.21.4",
]

[[package]]
name = "rustls"
version = "0.20.8"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "fff78fc74d175294f4e83b28343315ffcfb114b156f0185e9741cb5570f50e2f"
dependencies = [
 "log 0.4.17",
 "ring",
 "sct 0.7.0",
 "webpki 0.22.0",
]
```

Reply:
jfyi: `dangerous_configuration`: this feature enables a dangerous() method on ClientConfig and ServerConfig that allows setting inadvisable options, such as replacing the certificate verification process. Applications requesting this feature should be reviewed carefully.

assume this is for self-signed / local cert handling? cc @shamardy

Reply:
> assume this is for self-signed / local cert handling?

Yes. To disable certificate verification from cli side `dangerous_configuration` has to be used https://github.com/KomodoPlatform/komodo-defi-framework/blob/867a01ab04a57bab93ed52adf5729d165a43698a/mm2src/adex_cli/src/transport.rs#L106-L108
P.S. `dangerous_configuration` is not used from the https server side in mm2.

Reply:
Thank you for pointing it out )

[Originally](https://github.com/KomodoPlatform/komodo-defi-framework/commit/f8ada95e976cbebf5f680d13ec39d986a22e11d8#diff-1d929aa3bd497b52e68be0f0e96c4ad2a5506c5ce93e40953b5268ad976625a5R2236-R2246) the version of `rustls` was not constrained and "0.20.8" was used in `adex-cli`. 

<details>
  <summary>Version "0.19.1" is used as subdependency of `mm2_net`</summary>
  
```sh
$ cargo tree --manifest-path mm2src/adex_cli/Cargo.toml -i rustls@0.19.1
rustls v0.19.1
├── adex-cli v0.1.0 (/home/rozhkov/sources/atomicDEX-API/mm2src/adex_cli)
└── futures-rustls v0.21.1
    └── mm2_core v0.1.0 (/home/rozhkov/sources/atomicDEX-API/mm2src/mm2_core)
        └── mm2_net v0.1.0 (/home/rozhkov/sources/atomicDEX-API/mm2src/mm2_net)
            └── adex-cli v0.1.0 (/home/rozhkov/sources/atomicDEX-API/mm2src/adex_cli)
```
</details>

On 03.08 I had to start using `rustls` as explicit dependency and I was oriented on using `rustls` 0.20.4. It was a version which of mm2 [was dependent on](https://github.com/KomodoPlatform/komodo-defi-framework/blob/867a01ab04a57bab93ed52adf5729d165a43698a/Cargo.lock#L5998-L6022). Perhaps I had to strongly tie adex-cli on 0.20.4 to be able to manage versions manually.

Now using both version "0.19.1" and "0.20.8" looks appropriate in my honest opinion
---

Concerning the question why `dangerous_configuration` feature was utilized - that was to make it able to connect to `mm2` that issues self signed certificate.

Thank you

Reply:
> Concerning the question why dangerous_configuration feature was utilized - that was to make it able to connect to mm2 that issues self signed certificate.

mm2 can be initialized using a certificate file too, it doesn't have to be self-signed but will be in most cases. It would be good to allow the cli user to disable certificate verification themselves like it's done in some other clients (e.g. postman provides this, it doesn't disable it by default). Please open an issue for this @rozhkovdmitrii and it can be done later as it's not urgent at all.

Reply:
About dependency versions used in CLI, IMHO we should use same versions used in mm2 and update versions together. I already left a comment about this last week here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932#pullrequestreview-1572881800

Reply:
> About dependency versions used in CLI, IMHO we should use same versions used in mm2 and update versions together. I already left a comment about this last week here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932#pullrequestreview-1572881800

I will certainly polish all `cli` dependencies to fit them to mm2's. Thank you )

Reply:
>  Please open an issue for this @rozhkovdmitrii and it can be done later as it's not urgent at all.

[done](https://github.com/KomodoPlatform/komodo-defi-framework/issues/1942)

Reply:
I just wanted to mention something about the versions of `hyper-rustls` and `rustls`. From what I understand, the version of `rustls` that we use should exactly match the version of `rustls` that `hyper-rustls` depends on. For example, if we are using `hyper-rustls 0.23`, we should also use `rustls 0.20.8`. This is important because if there is a version mismatch, such as using `hyper-rustls 0.23` and `rustls 0.21.7`, we may encounter unexpected errors like below:
```
note: `ClientConfig` is defined in crate `rustls`
   --> /home/decker/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.21.7/src/client/client_conn.rs:128:1
    |
128 | pub struct ClientConfig {
    | ^^^^^^^^^^^^^^^^^^^^^^^
note: `rustls::client::client_conn::ClientConfig` is defined in crate `rustls`
   --> /home/decker/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.20.8/src/client/client_conn.rs:91:1
    |
91  | pub struct ClientConfig {
    | ^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `rustls` are being used?

```
To avoid any potential compatibility issues, it may be better to specify the exact versions of the crates using the `=` symbol. This way, we can ensure that the versions of `hyper-rustls` and `rustls` are precisely matched, reducing the chances of encountering any compatibility problems. Perhaps I may have slightly overestimated the significance of the "issue", but I have personally encountered package version mismatches during some of my own tests. Anyway, JFYI.

Reply:
Thank you for pointing it out :pray:, [solved](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/0f0dda9b61a7e255ccd3b29f24603126809b0a24)

Reply:
@rozhkovdmitrii can you please make this fix and the one here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1936#discussion_r1311983662 in a seperate PR? The whole release will be blocked until https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932 is sec reviewed and QA tested otherwise.

Reply:
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1956)